tests/test_pipelines.py

```diff
@@ -318,6 +318,138 @@ def test_tf_text_generation(self):
+                "candidate_labels": "politics",
+                "hypothesis_template": "Template without formatting syntax.",
+            },
+        ]
+        self.assertIsNotNone(nlp)
+
+        for mono_input in valid_mono_inputs:
+            mono_result = nlp(**mono_input)
+            self.assertIsInstance(mono_result, dict)
+            if len(mono_result["labels"]) > 1:
+                self._test_scores_sum_to_one(mono_result)
+
+            for key in output_keys:
+                self.assertIn(key, mono_result)
+
+        multi_result = nlp(**valid_multi_input)
+        self.assertIsInstance(multi_result, list)
+        self.assertIsInstance(multi_result[0], dict)
+        self.assertEqual(len(multi_result), len(valid_multi_input["sequences"]))
+
+        for result in multi_result:
+            for key in output_keys:
+                self.assertIn(key, result)
+
+            if len(result["labels"]) > 1:
+                self._test_scores_sum_to_one(result)
+
+        for bad_input in invalid_inputs:
+            self.assertRaises(Exception, nlp, **bad_input)
+
+    def _test_zero_shot_pipeline_outputs(self, nlp):
+        inputs = [
+            {
+                "sequences": "Who are you voting for in 2020?",
+                "candidate_labels": ["politics", "public health", "science"],
+            },
+            {
+                "sequences": "The dominant sequence transduction models are based on complex recurrent or convolutional neural networks in an encoder-decoder configuration. The best performing models also connect the encoder and decoder through an attention mechanism. We propose a new simple network architecture, the Transformer, based solely on attention mechanisms, dispensing with recurrence and convolutions entirely. Experiments on two machine translation tasks show these models to be superior in quality while being more parallelizable and requiring significantly less time to train. Our model achieves 28.4 BLEU on the WMT 2014 English-to-German translation task, improving over the existing best results, including ensembles by over 2 BLEU. On the WMT 2014 English-to-French translation task, our model establishes a new single-model state-of-the-art BLEU score of 41.8 after training for 3.5 days on eight GPUs, a small fraction of the training costs of the best models from the literature. We show that the Transformer generalizes well to other tasks by applying it successfully to English constituency parsing both with large and limited training data.",
+                "candidate_labels": ["machine learning", "statistics", "translation", "vision"],
+                "multi_class": True,
+            },
+        ]
+
+        expected_outputs = [
+            {
+                "sequence": "Who are you voting for in 2020?",
+                "labels": ["politics", "public health", "science"],
+                "scores": [0.9759403467178345, 0.015444041229784489, 0.00861559621989727],
```

Comment:
Does this overflow max_positions? Not an issue but maybe worthy of a comment.

Reply:
also this looks like lots of precision. Im sure it gets rounded above but 3 decimal places would be more readable.

Reply:
To make sure I understand: you're saying to just truncate the expected scores to 3 decimal places to make the test more readable?

Reply:
yes